front/app/components/UI/GanttChart/hooks/useGanttViewConfig.ts

```diff
@@ -0,0 +1,83 @@
+import { useMemo } from 'react';
+
+import {
+  groupDayCellsByMonth,
+  groupWeekCellsByYear,
+  groupQuarterCellsByMonth,
+  groupMonthCellsByYear,
+  getDayCells,
+  getWeekCells,
+  getQuarterCells,
+  getMonthCells,
+  getPreciseOffsetInDays,
+  getPreciseDurationInDays,
+  getPreciseOffsetInMonths,
+  getPreciseDurationInMonths,
+  getPreciseOffsetInWeeks,
+  getPreciseDurationInWeeks,
+  getPreciseOffsetInQuarters,
+  getPreciseDurationInQuarters,
+  quarterWidth,
+  weekWidth,
+  monthWidth,
+  dayWidth,
+} from '../utils';
+
+/**
+ * A custom hook that creates the view configuration object for the Gantt chart.
+ * This isolates the static view definitions from the main component's logic.
+ *
+ * @param timelineStartDate The start date of the visible timeline, used for calculations.
+ * @returns A memoized view configuration object.
+ */
+export const useGanttViewConfig = (timelineStartDate: Date) => {
+  const viewConfig = useMemo(
+    () => ({
+      month: {
+        unitWidth: dayWidth,
+        getGroups: (start: Date, end: Date) =>
+          groupDayCellsByMonth(getDayCells(start, end), () => dayWidth),
+        getFlatCells: getDayCells,
+        getOffset: (date: Date) =>
```

Comment:
One thing that was not immediately clear to me was what this 'offset' means. Maybe we could rename it to `getOffsetInDays`? That makes it immediately clear what this is for, and is also consistent with how `date-fns` calls it